woocommerce/woocommerce-android
# Feature/login local notifications

Part I: #6989


### Description
First iteration to schedule local notification to engage with users that don't interact with the app after install or experience some kind of error when trying to login. 

In this PR adds the following changes: 
- Infrastructure to be able to schedule local notifications after 24h using Android `WorkManager`
- Infrastructure to be able to inject dependencies to WorkManager "Workers" 
- Add new notification channel to send local notifications
- Notification canceling/dismissing when the user successfully logs in
- Re-schedule notification with the latest login error
- Control that once a "pre-login" notification is displayed the user won't see this type of notification again. 


This changes are hidden under a FEATURE_FLAG for debug builds. This won't have an impact on the app yet. Another PR with additional changes is on the way, where I'll remove the feature flag. 

### Testing instructions
- Open login flow
- Click login with site address and input a "non wordpress" site address. Like `google.com`
- Click continue
-  Wait around 5 secs for the notification to appear. Click on the notification and check it opens help/support section
- Repeat the same flow and check that the notification is not shown again

### Images/gif

https://user-images.githubusercontent.com/2663464/179853135-91b70044-db19-4f59-a8c1-978200d1f2d0.mp4
